cli/js/compiler.ts

```diff
@@ -81,259 +81,282 @@ interface CompileResult {
   diagnostics?: Diagnostic;
 }
 
-// TODO(bartlomieju): refactor this function into multiple functions
-// per CompilerRequestType
-async function tsCompilerOnMessage({
-  data: request
-}: {
-  data: CompilerRequest;
-}): Promise<void> {
-  switch (request.type) {
-    // `Compile` are requests from the internals to Deno, generated by both
-    // the `run` and `bundle` sub command.
-    case CompilerRequestType.Compile: {
-      const {
-        bundle,
-        config,
-        configPath,
-        outFile,
-        rootNames,
-        target
-      } = request;
-      util.log(">>> compile start", {
-        rootNames,
-        type: CompilerRequestType[request.type]
-      });
-
-      // When a programme is emitted, TypeScript will call `writeFile` with
-      // each file that needs to be emitted.  The Deno compiler host delegates
-      // this, to make it easier to perform the right actions, which vary
-      // based a lot on the request.  For a `Compile` request, we need to
-      // cache all the files in the privileged side if we aren't bundling,
-      // and if we are bundling we need to enrich the bundle and either write
-      // out the bundle or log it to the console.
-      const state: WriteFileState = {
-        type: request.type,
-        bundle,
-        host: undefined,
-        outFile,
-        rootNames
-      };
-      const writeFile = createWriteFile(state);
-
-      const host = (state.host = new Host({
-        bundle,
-        target,
-        writeFile
-      }));
-      let diagnostics: readonly ts.Diagnostic[] | undefined;
-
-      // if there is a configuration supplied, we need to parse that
-      if (config && config.length && configPath) {
-        const configResult = host.configure(configPath, config);
-        diagnostics = processConfigureResponse(configResult, configPath);
+type RuntimeCompileResult = [
+  undefined | DiagnosticItem[],
+  Record<string, string>
+];
+
+type RuntimeBundleResult = [undefined | DiagnosticItem[], string];
+
+async function requestCompile(
```

Comment:
seems to me this function would be better called "compile" rather than "requestCompile"